docs/adr/0002-schema-first-api-design.md

```diff
@@ -0,0 +1,59 @@
+# ADR 0002: Schema-First API Design
+
+## Status
+Voorstel
+
+## Context
+Om op een consistente en duidelijke manier een goed onderhoudbare contract voor de BRP API te kunnen ontwerpen en te kunnen documenteren, is er behoefte aan een werkwijze die dit faciliteert.
+
+## Beslissing
+We hebben besloten om het contract voor de BRP API te documenteren met behulp van de [OpenAPI Specificatie][OAS] en de [Schema-First API Design][SchemaFirst] aanpak te hanteren voor de BRP API ontwikkeling.
+Dit betekent dat voor elk te implementeren wijziging/toevoeging eerst wordt gekeken of de implementatie wijziging/toevoeging zal leiden tot een wijziging van het API contract. De wijziging aan het API contract wordt dan vóór de implementatie wijziging/toevoeging uitgevoerd.
+
+## Consequenties
+- **Voordelen:**
+    - Duidelijke en consistente API contract.
+    - Verbeterde communicatie tussen de stakeholders.
+    - Eerder en gemakkelijker valideren van het API contract.
+    - Betere Developer Experience (betere documentatie, client code generatie en consumers en provider van het API contract kunnen parallel worden geïmplementeerd).
+
+- **Nadelen:**
+    - Mogelijke leercurve voor stakeholders die niet bekend zijn met Schema-First API design.
+
+## Overwogen alternatieven
+- **Code-First API Design:** Eerst de wijziging implementeren en vervolgens het API contract genereren vanuit de code. Deze aanpak is niet gekozen omdat de kans groot is dat deze aanpak zal leiden tot onduidelijke en inconsistente API contracten en verkeerde aannames tijdens de implementatie.
+
+## Afspraken
+
+- Een externe API contract (wordt geconsumeerd door applicaties van zowel binnen als buiten de organisatie van de provider) wordt beheerd in een voor dat API opgezette GitHub repo. Een voorbeeld van zo'n repo is de [Haal Centraal BRP bevragen][ApiContractRepo] repo. De benodigde OAS bestanden van/voor het API contract en het gepubliceerde (in genereervariant map) OAS bestand van het API contract staan in de **specificatie** map van de repo.
+De repo waar de provider implementatie van het externe API contract wordt beheerd, bevat in zijn **specificatie** map van de **main** branch een kopie van dé versie van het OAS bestand van het externe API contract dat door de provider is geïmplementeerd. Een voorbeeld van zo'n repo is de [personen-informatie-service][InformatieServiceRepo] repo.
```

Comment:
duidelijk, wil je dit ook zo in Gezag? daar zit het nu src/resources/

Reply:
Is er een reden waarom dit nu in src/resources wordt beheerd? Zo niet, dan lijkt het mij wel handig om het in de specificatie map te stoppen. Dan kunnen de OAS scripts in de brp-shared repo worden hergebruikt. Ook is het dan makkelijker terug te vinden

Reply:
> De OAS is onderdeel van de source code ('/src'). Java source code komt in '/src/main' en de rest in '/src/resources'. In een standaard Spring Boot project kan het OAS bestand opgevraagd worden als static resource. Daarom is de algemene verwachting in Java/Spring land dat het OAS bestand te vinden is onder '/src/resources'.
> 
> In principe kunnen we hier van afwijken om uniformiteit na te streven binnen brp-api. Dat betekent wel dat de applicatie het OAS bestand niet meer kan serven als static resource. Dat is volgens mij ok binnen brp-api.

Helder.

We hebben ervoor gekozen om de OAS van een API niet te laten publiceren door zijn provider omdat het contract door meerdere providers kan worden geïmplementeerd. En omdat de brp-api niet zonder authenticatie en client certificaat kan worden bevraagd. Daarom wordt de OAS van de brp-api gepubliceerd in een publieke repo